src/stdlib_experimental_kinds.f90

```diff
@@ -0,0 +1,10 @@
+module stdlib_experimental_kinds
+! Instead of iso_fortran_env, we use iso_c_binding, to be compatible with C
+!use iso_fortran_env, only: sp=>real32, dp=>real64, qp=>real128
+!use iso_fortran_env, only: int32, int64, int128
+use iso_c_binding, only: sp=>c_float, dp=>c_double, qp=>c_float128
+use iso_c_binding, only: int32=>c_int32_t, int64=>c_int64_t, int128=>c_int128_t
```

Comment:
Is `int128` supported by all compilers?

Reply:
That's a good question. If C has standardized `int128` I would image that it is. But I need to consult the standard for more insight.

Reply:
At least I don't find `c_int128_t` in `iso_c_binding or `int128` in `iso_fortran_env`

Reply:
But it compiled. It might be only supported by gfortran?

Reply:
The standard says that it will compile, but be set to a special value when not supported (missing) if my memory is correct. So compilation doesn't guarantee existence of the kind. (I think the special value is 0 or a negative integer.)

Reply:
It looks like C99 has no guaranteed 128-bit integer (`long long` is "at least 64 bits" and `intN_t` is only guaranteed up to 64).

Looking at my own include files (libc 2.30), I don't see anything for 128-bit integers except a few places under an `__ILP32__` preprocessor flag, which must have been unset when my platform-specific includes were preprocessed for my system.

Reply:
Also, I think one does need to be careful with quad precision.  On its own, it does not have a clear meaning.  There are (at least) three active definitions:

https://gcc.gnu.org/onlinedocs/gcc/Floating-Types.html

For example, on my machine, `double` and `long double` both point to `_Float64`.  My libc also defines a `_Float64x` which as best I can tell corresponds to the x86 float80 format.

Reply:
BTW I am looking at a copy of the 2018 standard now (Table 18.2 in section 18.3.1), and I don't see any reference to `c_int128_t` in there?  Only up to `c_int64_t`.  Also no reference to `c_float128`.